x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -165,7 +165,17 @@ private static Map<String, RoleDescriptor> initializeReservedRoles() {
                     new String[] { "transport_client" },
                     null,
                     null,
-                    MetadataUtils.DEFAULT_RESERVED_METADATA
+                    null,
+                    null,
+                    MetadataUtils.DEFAULT_RESERVED_METADATA,
+                    null,
+                    null,
+                    null,
+                    null,
+                    "Grants the privileges required to access the cluster through the Java Transport Client. "
+                        + "The Java Transport Client fetches information about the nodes in the cluster using "
+                        + "the Node Liveness API and the Cluster State API (when sniffing is enabled). "
+                        + "Assign your users this role if they use the Transport Client."
```

Comment:
not finding the node liveness api documented anywhere.

does "when sniffing is enabled" modify both the node liveness API + cluster state api? if so:

```suggestion
                    "Grants the privileges required to access the cluster through the Java Transport Client. "
                        + "When sniffing is enabled, the Java Transport Client can fetch information about the nodes in the cluster using "
                        + "the Node Liveness API and the Cluster State API. "
                        + "Assign your users this role if they use the Transport Client."
```
Additionally, is it the user that is using the JTC, or is the user being used to auth the JTC?

to clarify, you're intending to refer to the [deprecated java transport client](https://www.elastic.co/guide/en/elasticsearch/client/java-api/current/transport-client.html), and not the newest java client, correct?

I see that you basically just copied the text from the [original description](https://www.elastic.co/guide/en/elasticsearch/reference/current/built-in-roles.html) so you can take this all with a grain of salt.

Reply:
> not finding the node liveness api documented anywhere.

Yes, the liveness action doesn't exist anymore.

Reply:
> does "when sniffing is enabled" modify both the node liveness API + cluster state api?

Just checked this and the Transport Client is periodically checking if the configured list of nodes is alive. Even without sniff. So I will leave this description as is.